File: src/main/java/io/baroka/exception/payload/ExceptionMsg.java
package io.baroka.exception.payload;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import lombok.*;

import java.io.Serializable;
import java.util.List;


@RequiredArgsConstructor
@AllArgsConstructor
@Data
@Builder
public final class ExceptionMsg implements Serializable {
	
	private static final long serialVersionUID = 1L;

	
	@JsonInclude(value = Include.NON_NULL)
	private Throwable throwable;
	
	private final String msg;

	private final int code;

	private final  boolean success;

	private List<FieldErrorDetail> errors;

	@Builder
	@Getter
	@Setter
	public static class FieldErrorDetail {
		private String field;
		private String message;
	}
}

File: src/main/java/io/baroka/exception/payload/ExceptionMsg.kt
package io.baroka.exception.payload

import groovy.transform.builder.Builder
import java.io.Serializable

@Builder
data class ExceptionMsg (
    val msg: String,
    val code: Int,
    val success: Boolean,
    val errors: List<FieldErrorDetail>
) : Serializable {

    companion object {
        private const val serialVersionUID = 1L
    }

    constructor(msg: String, code: Int, success: Boolean) : this(msg, code, success, emptyList())

}

data class FieldErrorDetail(
    var field: String,
    var messag: String
)